internal/provider/resource_vault_cluster_admin_token.go

```diff
@@ -42,18 +54,160 @@ func resourceVaultClusterAdminToken() *schema.Resource {
+		Provider: cluster.Location.Region.Provider,
+		Region:   cluster.Location.Region.Region,
+	}
+
+	tokenResp, err := clients.CreateVaultClusterAdminToken(ctx, client, loc, clusterID)
+	if err != nil {
+		return diag.Errorf("error creating HCP Vault cluster admin token (cluster_id %q) (project_id %q): %+v",
+			clusterID,
+			client.Config.ProjectID,
+			err,
+		)
+	}
+
+	err = d.Set("token", tokenResp.Token)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	err = d.Set("created_at", time.Now().Format(time.RFC3339))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(fmt.Sprintf("/project/%s/%s/%s/token",
+		loc.ProjectID,
+		VaultClusterResourceType,
+		clusterID))
+
 	return nil
 }
 
-// resourceVaultClusterAdminTokenRead will act as a no-op as the admin token is not persisted in
-// any way that it can be fetched and read
-func resourceVaultClusterAdminTokenRead(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+// resourceVaultClusterAdminTokenRead cannot read the admin token from the API as it is not persisted in
+// any way that it can be fetched. Instead this operation first verifies the existence of the associated Vault cluster
+// and then refreshes the token if it is close to expiring or expired.
+func resourceVaultClusterAdminTokenRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*clients.Client)
+
+	clusterID := d.Get("cluster_id").(string)
+	organizationID := client.Config.OrganizationID
+	projectID := client.Config.ProjectID
+
+	loc := &models.HashicorpCloudLocationLocation{
+		OrganizationID: organizationID,
+		ProjectID:      projectID,
+	}
+
+	log.Printf("[INFO] reading Vault cluster (%s) [project_id=%s, organization_id=%s]", clusterID, loc.ProjectID, loc.OrganizationID)
+
+	cluster, err := clients.GetVaultClusterByID(ctx, client, loc, clusterID)
+	if err != nil {
+		if clients.IsResponseCodeNotFound(err) {
+			// No cluster exists, so this admin token should be removed from state.
+			log.Printf("[WARN] no HCP Vault cluster found with (cluster_id %q) (project_id %q); removing admin token.",
+				clusterID,
+				projectID,
+			)
+			d.SetId("")
+			return nil
+		}
+
+		return diag.Errorf("unable to check for presence of an existing Vault cluster (cluster_id %q) (project_id %q): %v",
+			clusterID,
+			projectID,
+			err,
+		)
+	}
+
+	loc.Region = &models.HashicorpCloudLocationRegion{
+		Provider: cluster.Location.Region.Provider,
+		Region:   cluster.Location.Region.Region,
+	}
+
+	// If the token already exists, this block verifies if it is close to expiration and should be refreshed.
+	createdAt := d.Get("created_at").(string)
+	if createdAt != "" {
+		log.Printf("[INFO] existing admin token found for Vault cluster (%s) [project_id=%s, organization_id=%s]",
+			clusterID,
+			loc.ProjectID,
+			loc.OrganizationID,
+		)
+
+		// The refresh window starts five minutes before the 6h expiry.
```

Comment:
Awesome, I was going to suggest this exact thing if it wasn't already implemented 🙌